sdk/core/Azure.Core/src/SerializableOptions.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure
+{
+    /// <summary>
+    /// TODO
+    /// </summary>
+    public class SerializableOptions
```

Comment:
It might be helpful to look at the System.Text.Json serialization-related options types, and understand if there are options we would want to support from those and/or if we would want to design the API to get those options from related types like @heaths suggested here: https://github.com/Azure/azure-sdk-for-net/pull/35742/files#r1176840779

- [JsonSerializerOptions](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializeroptions?view=net-7.0)
- [JsonWriterOptions](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonwriteroptions?view=net-8.0)
- [JsonReaderOptions](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonreaderoptions?view=net-8.0)

Reply:
Our options are purposefully more restrictive here since the idea is we already have all the renaming / restructuring logic written internally and we simply want to expose this to the customers publicly.  We don't really want to expose the ability for them to further override these settings since then they aren't getting a universally transferable format.  It would even be difficult to serialize / deserialize within the same library if they could override these types of things.

If there are any specific options you think should be included on top of what we have please let us know.

Reply:
What do you mean differently-formatted JSON is not "universally transferrable"? JSON parsers are supposed to ignore insignificant whitespace by design. I've never seen one that doesn't across a multitude of languages and class libraries. It wouldn't be a true JSON parser otherwise.

Our definition of "pretty print" will vary from person to person / use to use. We should make it at least as robust as STJ or consider not supporting it until there's need.

Reply:
I was referring to the other options which allowed you to define name changes etc through custom json converters.  We want to avoid exposing these since we are pinning ourselves to Azure Wire Format where we can control the translation layer so that other languages can deserialize the objects we serialize.

We have played with IAzureJsonSerializable and AzureJsonSerializableOptions to make this point clear but avoided that in this first draft.

PrettyPrint I agree this is a universally accepted concept.

Reply:
I was referring to preferences like how many spaces to indent, or even to use tabs instead of spaces. These are things that STJ and even Newtonsoft.Json expose. `PrettyPrint` is doing a lot of heavy lifting - forcing a lot of our preferences onto customers.